src/moscot/problems/base/_mixins.py

```diff
@@ -476,3 +477,93 @@ def _cell_aggregation_transition(
+        seed: Optional[int] = None,
+        **kwargs: Any,
+    ) -> pd.DataFrame:
+        """
+        Compute correlation of push or pull distribution with features.
+
+        Correlates (processed) count data with probabilities of cells mapped to a subset of cells, e.g. for a
+        certain celltype.
+
+        Parameters
+        ----------
+        key
+            Column of :attr:`adata.obs` containing push-forward or pull-back distributions.
+        method
+            Mode to use when calculating p-values and confidence intervals. Valid options are:
+                - `{TestMethod.FISCHER!r}` - use Fischer transformation :cite:`fischer:21`.
+                - `{TestMethod.PERM_TEST!r}` - use permutation test.
+        annotation
+            If not `None`, this defines the subset of data to be considered. The key of the :class:`dict` should
+            correspond to a column in :attr:`anndata.AnnData.obs` while the value should be a list of values in this
+            column.
+        layer
+            Key from :attr:`anndata.AnnData.layers` from which to get the expression.
+            If `None` or `'X'`, use :attr:`anndata.AnnData.X`.
+        features
+            A subset of :attr:`anndata.AnnData.var` the features the distribution in :attr:`anndata.AnnData.obs[]`
+            should be compared to. If `None`, all features will be taken into account.
+        confidence_level
+            Confidence level for the confidence interval calculation. Must be in interval `[0, 1]`.
+        n_perms
+            Number of permutations to use when ``method = {TestMethod.PERM_TEST!r}``.
+        seed
+            Random seed when ``method = {TestMethod.PERM_TEST!r}``.
+
+        %(parallel_kwargs)s
+
+        Returns
+        -------
+        %(correlation_test.returns)s
+        """
+        if key not in self.adata.obs.columns:
+            raise KeyError(f"[{key!r} not found in `adata.obs`.")
+
+        method = TestMethod(method)
+
+        if annotation is not None:
+            annotation_key, annotation_vals = next(iter(annotation.items()))  # type:ignore[misc]
+            if annotation_key not in self.adata.obs.columns:  # type:ignore[has-type]
+                raise KeyError(f"[{annotation_key!r} not found in `adata.obs`.")  # type:ignore[has-type]
+            if not isinstance(annotation_vals, list):  # type:ignore[has-type]
+                raise TypeError("`annotation` expected to be dictionary of length 1 with value being a list.")
+
+            adata_red = self.adata[self.adata.obs[annotation_key].isin(annotation_vals)]  # type:ignore[has-type]
+        else:
+            adata_red = self.adata
+
+        adata_red = adata_red[~adata_red.obs[key].isnull()]
+        if len(adata_red) == 0:
+            raise ValueError(f"`adata.obs[{key!r}]` only contains NaN values.")
+        distribution = adata_red.obs[[key]]
+
+        if features is not None:
+            adata_red = adata_red[:, features]
```

Comment:
I would use `sc.get.obs_df` so that you can be flexible with either features or obs

Reply:
but this would only give the `pd.DataFrame`, right

Reply:
yes but I think that'd be ok cause then you have the flexibility of taking also continuous variables from obs